### ISSUE TITLE
retrain command

### ISSUE THREAD

**Issue by liao1123** [locuslab/open-unlearning#84]
Hello, thank you very much for open-sourcing the code. I'm new to LLM unlearning. However, I have some questions:

I'm a bit confused about the retrain command. It seems to use the retained dataset for training and is similar to the fine-tuning command, but does it specify the dataset in the end? Could you provide an example?

Additionally, I find the "forget_quality" metric does not appear in my final JSONL results. How should this be calculated?

I'm trying to run the command line to achieve results similar to the benchmark, but the performance is somewhat poor. What is the expected margin of error?

**Comment by Dornavineeth** [comment]
Hi. Thank you for using our repository!

> Additionally, I find the "forget_quality" metric does not appear in my final JSONL results. How should this be calculated?

The forget quality of a retain model is 0 by definition. Forget quality for an unlearned/finetuned model is calculated with respect to retain model.  Please check [section 2.2.3](https://arxiv.org/pdf/2401.06121)

> I'm a bit confused about the retrain command. It seems to use the retained dataset for training and is similar to the fine-tuning command, but does it specify the dataset in the end? Could you provide an example?

We do specify the dataset, compare [line 35](https://github.com/locuslab/open-unlearning/blob/4d81a08430e999b70024b145f7bae32c2ebbc537/scripts/tofu_finetune.sh#L35)(for retain model) and [line 61](https://github.com/locuslab/open-unlearning/blob/4d81a08430e999b70024b145f7bae32c2ebbc537/scripts/tofu_finetune.sh#L61) (for full fine-tuned model). The datasets are set to retain and full splits, respectively.

> I'm trying to run the command line to achieve results similar to the benchmark, but the performance is somewhat poor. What is the expected margin of error?

Can you please share the exact commands you ran? Please note that using different hyperparams such as effective batch size, lr, etc, and even using different GPUs with accelerate can cause different results.

Hope this clarifies your query.

**Comment by liao1123** [comment]
> Hi. Thank you for using our repository!
> 
> > Additionally, I find the "forget_quality" metric does not appear in my final JSONL results. How should this be calculated?
> 
> The forget quality of a retain model is 0 by definition. Forget quality for an unlearned/finetuned model is calculated with respect to retain model. Please check [section 2.2.3](https://arxiv.org/pdf/2401.06121)
> 
> > I'm a bit confused about the retrain command. It seems to use the retained dataset for training and is similar to the fine-tuning command, but does it specify the dataset in the end? Could you provide an example?
> 
> We do specify the dataset, compare [line 35](https://github.com/locuslab/open-unlearning/blob/4d81a08430e999b70024b145f7bae32c2ebbc537/scripts/tofu_finetune.sh#L35)(for retain model) and [line 61](https://github.com/locuslab/open-unlearning/blob/4d81a08430e999b70024b145f7bae32c2ebbc537/scripts/tofu_finetune.sh#L61) (for full fine-tuned model). The datasets are set to retain and full splits, respectively.
> 
> > I'm trying to run the command line to achieve results similar to the benchmark, but the performance is somewhat poor. What is the expected margin of error?
> 
> Can you please share the exact commands you ran? Please note that using different hyperparams such as effective batch size, lr, etc, and even using different GPUs with accelerate can cause different results.
> 
> Hope this clarifies your query.

Thank you very much for answering my questions. I suddenly see the light!

**Comment by Dornavineeth** [comment]
Closing this issue as it is resolved. Happy unlearning!
